packages/ai-codex/src/browser/renderers/command-execution-renderer.tsx

```diff
@@ -0,0 +1,107 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { ChatResponseContent } from '@theia/ai-chat';
+import { ChatResponsePartRenderer } from '@theia/ai-chat-ui/lib/browser/chat-response-part-renderer';
+import { codicon } from '@theia/core/lib/browser';
+import { injectable } from '@theia/core/shared/inversify';
+import * as React from '@theia/core/shared/react';
+import { ReactNode } from '@theia/core/shared/react';
+import { CommandExecutionItem } from '../../common';
+import { CodexToolCallChatResponseContent } from '../codex-tool-call-content';
+import { nls } from '@theia/core';
+
+@injectable()
+export class CommandExecutionRenderer implements ChatResponsePartRenderer<CodexToolCallChatResponseContent> {
+    canHandle(response: ChatResponseContent): number {
+        return response.kind === 'toolCall' &&
+            (response as CodexToolCallChatResponseContent).name === 'command_execution'
```

Comment:
We should make this typesafe with an instanceof check or a marker property (see `ClaudeCodeToolCallChatResponseContent`).